### PR TITLE
feat: Migrate Async Storage v1 to LEGACY

### PR DESCRIPTION
**Pull request by Krizzu**
Summary:
---------

This is going to be the last release of v1 from the `master` branch. 
`LEGACY` branch will be the main source, while `master` will be the v2 place.
We'll keep applying fixes/features, until v2 releases.

### CODE REVIEW

**Comment by Krizzu** [comment]
cc @tido64 
Let's merge this :)

File: .circleci/config.yml [hunk 1]
```diff
@@ -277,14 +277,6 @@ jobs:
           name: Run e2e tests
           command: yarn test:e2e:android
 
-  "Publish: semantic release":
-    <<: *js_defaults
-    steps:
-      - *addWorkspace
-      - run:
-          name: Publish to NPM registry
-          command: yarn ci:publish
```
Comment (tido64) [inline .circleci/config.yml]: If you remove this job, how is this going to be the last release? The previous one will be the last.

Comment (Krizzu) [reply .circleci/config.yml]: I meant to release v1.5 manually

**Comment by Krizzu** [comment]
Thanks!

**Pull request merged**